PyQtGraph colour menu changes upset adding the Auto action

### Summary

There is a change in PyQtGraph between 0.13.3 and 0.13.4 in the colourbar context menu.

We currently insert the "Auto" item in the 12 position, but now that the menu is shorter this triggers a
`IndexError: list index out of range`

The old menu looked like
![image](https://github.com/mantidproject/mantidimaging/assets/74248560/1ab76b6b-eb5e-4e3b-bc04-ecce86dd4d81)


The new menu (with adding the Auto item)
![image](https://github.com/mantidproject/mantidimaging/assets/74248560/91532dcf-27e8-4b50-b2bc-2646be52b7a9)


### Steps To Reproduce

Update pyqtgraph
`mamba install pyqtgraph=0.13.4`

Open a dataset and right click on the colour bar

### Expected Behaviour

Menu to choose colours, with the Auto item

### Current Behaviour

`IndexError: list index out of range`

### Context

Current main.

Our conda/meta.yaml has `pyqtgraph=0.13.*` so we started picking up the new version today.

### Failure Logs
```
self = <mantidimaging.gui.widgets.mi_image_view.view.MIImageView object at 0x7f1a95258310>
parent = <mantidimaging.gui.widgets.mi_image_view.view.MIImageView object at 0x7f1a95258310>, recon_mode = False, index = 12, set_enabled = True

    def add_auto_color_menu_action(self,
                                   parent: 'Optional[QWidget]',
                                   recon_mode: bool = False,
                                   index: int = DEFAULT_MENU_POSITION,
                                   set_enabled: bool = True) -> QAction:
        self.auto_color_parent = parent
        self.auto_color_recon_mode = recon_mode
        self.auto_color_action = QAction("Auto")
>       place = self.histogram.gradient.menu.actions()[index]
E       IndexError: list index out of range

mantidimaging/gui/widgets/auto_colour_menu/auto_color_menu.py:46: IndexError

```

--- a/conda/meta.yaml
+++ b/conda/meta.yaml
@@ -36,7 +36,7 @@ requirements:
     - ccpi-regulariser
     - jenkspy=0.2.0
     - pyqt=5.15.*
-    - pyqtgraph=0.13.3
+    - pyqtgraph=0.13.7
     - qt-material=2.14
     - darkdetect=0.8.0
     - qt-gtk-platformtheme # [linux]

--- a/docs/release_notes/next/fix-2119-pyqt-menu
+++ b/docs/release_notes/next/fix-2119-pyqt-menu
@@ -1,0 +1,1 @@
+#2129 : fix menu position for new pyqtgraph

--- a/mantidimaging/gui/widgets/auto_colour_menu/auto_color_menu.py
+++ b/mantidimaging/gui/widgets/auto_colour_menu/auto_color_menu.py
@@ -12,8 +12,6 @@ if TYPE_CHECKING:
     import numpy as np
     from PyQt5.QtWidgets import QWidget
 
-DEFAULT_MENU_POSITION = 12
-
 
 class AutoColorMenu:
     """
@@ -38,12 +36,13 @@ class AutoColorMenu:
     def add_auto_color_menu_action(self,
                                    parent: QWidget | None,
                                    recon_mode: bool = False,
-                                   index: int = DEFAULT_MENU_POSITION,
                                    set_enabled: bool = True) -> QAction:
         self.auto_color_parent = parent
         self.auto_color_recon_mode = recon_mode
         self.auto_color_action = QAction("Auto")
-        place = self.histogram.gradient.menu.actions()[index]
+
+        index_of_rgb_item = [action.text() for action in self.histogram.gradient.menu.actions()].index("RGB")
+        place = self.histogram.gradient.menu.actions()[index_of_rgb_item - 1]
 
         self.histogram.gradient.menu.insertAction(place, self.auto_color_action)
         self.histogram.gradient.menu.insertSeparator(self.auto_color_action)

